cmd/telemetrygen/Dockerfile

```diff
@@ -1,4 +1,4 @@
-FROM golang:1.20 AS build
+FROM golang:1.20.7 AS build
```

Comment:
This is not needed

Reply:
I think the correct conversion here is to change it to 1.21
ref: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/18436/files?file-filters%5B%5D=No+extension&show-viewed-files=true#diff-15a455599a44216f0cc2d8d039375699ab5408b90dad03fff2f5040bcadb3f84L1

Reply:
looking at older [PR](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/18436/files?file-filters%5B%5D=No+extension&show-viewed-files=true#diff-15a455599a44216f0cc2d8d039375699ab5408b90dad03fff2f5040bcadb3f84L1)
Yes, this is not needed, I think all the `Dockerfile` changes can be reverted, added a commit.